plugin/advancedsubscription/ajax/advsub.ajax.php

```diff
@@ -0,0 +1,248 @@
+$data['is_connected'] = isset($_REQUEST['is_connected']) ? boolval($_REQUEST['is_connected']) : false;
+$data['profile_completed'] = isset($_REQUEST['profile_completed']) ? floatval($_REQUEST['profile_completed']) : 0;
+// Init result array
+$result = array('error' => true, 'errorMessage' => get_lang('ThereWasAnError'));
+// Check if data is valid or is for start subscription
+$verified = $plugin->checkHash($data, $hash) || $data['a'] == 'subscribe';
+if ($verified) {
+    switch($data['a']) {
+        case 'check': // Check minimum requirements
+            try {
+                $res = AdvancedSubscriptionPlugin::create()->isAbleToRequest($data['u'], $data);
+                if ($res) {
+                    $result['error'] = false;
+                    $result['errorMessage'] = 'No error';
+                    $result['pass'] = true;
+                } else {
+                    $result['errorMessage'] = 'User can not be subscribed';
+                    $result['pass'] = false;
+                }
+            } catch (\Exception $e) {
+                $result['errorMessage'] = $e->getMessage();
+            }
+            break;
+        case 'subscribe': // Subscription
+            // Start subscription to queue
+            $res = AdvancedSubscriptionPlugin::create()->startSubscription($data['u'], $data['s'], $data);
+            // Check if queue subscription was successful
+            if ($res === true) {
+                // Prepare data
+                // Get session data
+                $sessionArray = api_get_session_info($data['s']);
+                $extraSession = new ExtraFieldValue('session');
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'description');
+                $sessionArray['description'] = $var['field_valiue'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'target');
+                $sessionArray['target'] = $var['field_valiue'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'mode');
+                $sessionArray['mode'] = $var['field_valiue'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'publication_end_date');
+                $sessionArray['publication_end_date'] = $var['field_value'];
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'recommended_number_of_participants');
+                $sessionArray['recommended_number_of_participants'] = $var['field_valiue'];
+                // Get student data
+                $studentArray = api_get_user_info($data['u']);
+                $studentArray['picture'] = UserManager::get_user_picture_path_by_id($studentArray['user_id'], 'web', false, true);
+                $studentArray['picture'] = UserManager::get_picture_user($studentArray['user_id'], $studentArray['picture']['file'], 22, USER_IMAGE_SIZE_MEDIUM);
```

Comment:
For user picture :
First function, return an array with dir and filename for user picture
Second function, return an array with style and complete filename for medium size,
This medium size is needed for a uniform list of student. Then is needed second function, but second function require user picture filename, then it needs first function too. I dont know if exist a function to get medium size user picture using only user id